src/routes/transactions/mappers/transfers/transfer-info.mapper.ts

```diff
@@ -0,0 +1,105 @@
+import { Inject, Injectable } from '@nestjs/common';
+import { Safe } from '../../../../domain/safe/entities/safe.entity';
+import {
+  ERC20Transfer,
+  ERC721Transfer,
+  NativeTokenTransfer,
+  Transfer,
+} from '../../../../domain/safe/entities/transfer.entity';
+import { Token } from '../../../../domain/tokens/entities/token.entity';
+import { TokenRepository } from '../../../../domain/tokens/token.repository';
+import { ITokenRepository } from '../../../../domain/tokens/token.repository.interface';
+import { AddressInfoHelper } from '../../../common/address-info/address-info.helper';
+import { TransferTransactionInfo } from '../../entities/transfer-transaction-info.entity';
+import { Erc20Transfer } from '../../entities/transfers/erc20-transfer.entity';
+import { Erc721Transfer } from '../../entities/transfers/erc721-transfer.entity';
+import { NativeCoinTransfer } from '../../entities/transfers/native-coin-transfer.entity';
+import { TransferDirectionHelper } from '../common/transfer-direction.helper';
+
+@Injectable()
+export class TransferInfoMapper {
+  private static readonly ERC20_TRANSFER = 'ERC20_TRANSFER';
+  private static readonly ERC721_TRANSFER = 'ERC721_TRANSFER';
+  private static readonly ETHER_TRANSFER = 'ETHER_TRANSFER';
+
+  constructor(
+    @Inject(ITokenRepository) private readonly tokenRepository: TokenRepository,
+    private readonly addressInfoHelper: AddressInfoHelper,
+    private readonly transferDirectionHelper: TransferDirectionHelper,
+  ) {}
+
+  async mapTransferInfo(
+    chainId: string,
+    transfer: Transfer,
+    safe: Safe,
+  ): Promise<TransferTransactionInfo> {
+    const { from, to, type } = transfer;
+    const sender = await this.addressInfoHelper.getOrDefault(chainId, from);
+    const recipient = await this.addressInfoHelper.getOrDefault(chainId, to);
+    const direction = this.transferDirectionHelper.getTransferDirection(
+      safe.address,
+      from,
+      to,
+    );
+
+    switch (type) {
+      case TransferInfoMapper.ERC20_TRANSFER: {
+        const { tokenAddress, value } = transfer as ERC20Transfer;
+        const token = await this.getToken(chainId, tokenAddress);
+        return new TransferTransactionInfo(
+          sender,
+          recipient,
+          direction,
```

Comment:
Given that we always build a `TransferTransactionInfo` with the same `sender`, `recipient` and `direction` maybe we can return it in the end and use the switch/case just to build the different transfers wdyt?

Reply:
I agree, I think it's more clear that way. I refactored the main function in the class to have just one exit point and also moved the `Transfer` mapping to a separate function `getTransferByType` 🙂 Let me know if you find it more readable this way.

Reply:
Refactored in [2f6d0c5](https://github.com/5afe/safe-client-gateway-nest/pull/206/commits/2f6d0c508eace06c134039e921fe51865fc195eb)